std/math/big/int.zig

```diff
@@ -0,0 +1,2023 @@
+const std = @import("../../index.zig");
+const builtin = @import("builtin");
+const debug = std.debug;
+const math = std.math;
+const mem = std.mem;
+const Allocator = mem.Allocator;
+const ArrayList = std.ArrayList;
+
+const TypeId = builtin.TypeId;
+
+pub const Limb = usize;
+pub const DoubleLimb = @IntType(false, 2 * Limb.bit_count);
+pub const Log2Limb = math.Log2Int(Limb);
+
+comptime {
+    debug.assert(math.floorPowerOfTwo(usize, Limb.bit_count) == Limb.bit_count);
+    debug.assert(Limb.bit_count <= 64); // u128 set is unsupported
+    debug.assert(Limb.is_signed == false);
+}
+
+const wrapped_buffer_size = 512;
+
+// Converts primitive integer values onto a stack-based big integer, or passes through existing
+// Int types with no modifications. This can fail at runtime if using a very large dynamic
+// integer but it is very unlikely and is considered a user error.
+fn wrapInt(allocator: *Allocator, bn: var) *const Int {
```

Comment:
The reason I have this to be unconditional (not return an error) is that something like `cmp` should work without error. This does mean there is an upper limit on comptime integers passed as arguments. Fine in practice, a user can simply call `Bn.set` first in this case.

The `wrapper_buffer_size` is larger than it needs to be as well. Technically `512` gives us space for `512 * 8 = 4096` bit comptime arguments, but the maximum size is actually smaller `512`. Probably just overlooked something or the allocator is over-allocating (shouldn't be).

Reply:
I think this makes sense. We should be able to get a compile error (not runtime error) when the value cannot fit, right?

Reply:
If it starts to get too complicated, we can always cast the `wrapInt` value to `u64`/`i64` and then document that one should use `Bn.set` for values outside this range.